src/shogun/lib/SGVector.cpp

```diff
@@ -108,47 +114,57 @@ SGVector<T>::~SGVector()
 
 template <class T>
 SGVector<T>::SGVector(EigenVectorXt& vec)
-: SGReferencedData(false), vector(vec.data()), vlen(vec.size())
+: SGReferencedData(false), vector(vec.data()), vlen(vec.size()), gpu_vector(NULL)
 {
 
 }
 
 template <class T>
 SGVector<T>::SGVector(EigenRowVectorXt& vec)
-: SGReferencedData(false), vector(vec.data()), vlen(vec.size())
+: SGReferencedData(false), vector(vec.data()), vlen(vec.size()), gpu_vector(NULL)
 {
 
 }
 
 template <class T>
 SGVector<T>::operator EigenVectorXtMap() const
 {
+	if (on_gpu())
```

Comment:
THen you just put a single method call everywhere where you dont want people to do things